app/decorators/controllers/solidus_marketplace/spree/admin/reports_controller_decorator.rb

```diff
@@ -57,11 +103,11 @@ def get_supplier_earnings
         def grouped_supplier_earnings
           params[:q] = search_params
 
-          @search = Spree::Order.complete.not_canceled.ransack(params[:q])
+          @search = ::Spree::Order.complete.not_canceled.ransack(params[:q])
           @orders = @search.result
 
           supplier_earnings_map = @orders.map(&:supplier_earnings_map)
-          grouped_suppliers_map = supplier_earnings_map.flatten.group_by(&:name).values
+          grouped_suppliers_map = supplier_earnings_map.flatten.group_by { |s| s[:name] }.values
```

Comment:
here in case that the suppliers aren't present, maybe we should use:

```
supplier_earnings_map.flatten.group_by { |s| s[:name] }&.values
```

Reply:
Let me test that first and then change it